main.cpp

```diff
@@ -35,9 +36,9 @@ int main(int argc, char **argv)
 		}
 	}
 
-	GameServer gameServer;
+	std::unique_ptr<GameServer> gameServer;

```

Comment:
The local variable can be constructed with a parameter list. That would remove the need for late construction and the whole `std::unique_ptr` and `<memory>` usage. A local variable is cleaned up automatically when `main` exits.

```cpp
GameServer gameServer(portNum);
```

----

Rather than `return`, the `try` block should perhaps encompass more of the code, covering the whole area where the local variable is used.

```cpp
try {
    GameServer gameServer(portNum);
    // ...
    for (;;) {
        // ...
    }
} catch(const std::exception& e) {
    // ...
}
```

Reply:
Oh yeah, why didn't I think about that.